cuda_core/cuda/core/experimental/_event.pyx

```diff
@@ -182,16 +193,20 @@ cdef class Event:
     def from_ipc_descriptor(cls, ipc_descriptor: IPCEventDescriptor) -> Event:
         """Import an event that was exported from another process."""
         cdef cydriver.CUipcEventHandle data
+        cdef cydriver.CUcontext curr_ctx
+        cdef cydriver.CUdevice dev
         memcpy(data.reserved, <const void*><const char*>(ipc_descriptor._reserved), sizeof(data.reserved))
         cdef Event self = Event.__new__(cls)
         with nogil:
             HANDLE_RETURN(cydriver.cuIpcOpenEventHandle(&self._handle, data))
+            HANDLE_RETURN(cydriver.cuCtxGetCurrent(&curr_ctx))
+            HANDLE_RETURN(cydriver.cuCtxGetDevice(&dev))
```

Comment:
I feel this is problematic, unless we can gain internal knowledge and be 100% certain. (I also left a comment when this was checked in, but it was not addressed.) 

We should send the device ID over to the child process too. Unfortunately unlike streams there is no way for us to do a reverse lookup of the device that an event was created on, and so we need to do bookkeeping ourselves and make use of it. There is no guarantee that the current device when `from_ipc_descriptor` is called is the same as the one on which the event was created.